superset/common/query_context.py

```diff
@@ -101,21 +104,143 @@ def __init__(  # pylint: disable=too-many-arguments
             "result_format": self.result_format,
         }
 
-    def get_query_result(self, query_object: QueryObject) -> Dict[str, Any]:
-        """Returns a pandas dataframe based on the query object"""
+    @staticmethod
+    def left_join_on_dttm(
+        left_df: pd.DataFrame, right_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        df = left_df.set_index(DTTM_ALIAS).join(right_df.set_index(DTTM_ALIAS))
+        df.reset_index(level=0, inplace=True)
+        return df
+
+    def processing_time_offsets(
```

Comment:
NIT: looking at the function signature, personally I'm not big on methods that return a tuple of mixed objects, it feels like maybe we should be mutating the object this method is part of, or returning a more complex object with a proper name "CachedDataFrame" or something (unclear to me at this point what the best name would be). I namedtuple could be a good compromise.

I'm not sure what it implies as I'm just starting to review this code

Reply:
I'm going to do a little refactoring for the return value(Returns an object for this function). Thanks for the heads up.